scripts/ci/verify_browser_runtime_artifact.py

```
@@ -0,0 +1,287 @@
+
+    actual_sha = _git_sha(source_root)
+    if expected_sha:
+        normalized_expected = expected_sha.strip()
+        if not _SHA_RE.fullmatch(normalized_expected):
+            raise VerificationError(f"expected source SHA is not a full commit SHA: {expected_sha!r}")
+        if actual_sha is None:
+            raise VerificationError("could not resolve the checked-out source SHA")
+        if actual_sha.lower() != normalized_expected.lower():
+            raise VerificationError(
+                f"artifact source SHA mismatch: expected {normalized_expected}, got {actual_sha}"
+            )
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)